x/wasm/keeper/keeper.go

```diff
@@ -299,6 +311,11 @@ func (k Keeper) instantiate(ctx sdk.Context, codeID uint64, creator, admin sdk.A
 		return nil, nil, sdkerrors.Wrap(err, "dispatch")
 	}
 
+	ctx.EventManager().EmitEvent(sdk.NewEvent(
```

Comment:
Please emit this *before* calling `k.wasmVM.Instantiate`.

We have all needed info there, and this is the ordering from the spec.